source/slang/memory_pool.h

```diff
@@ -0,0 +1,19 @@
+#ifndef SLANG_MEMORY_POOL_H
+#define SLANG_MEMORY_POOL_H
+
+#include "../core/basic.h"
+
+namespace Slang
+{
+    struct MemoryPoolSegment;
+
+    struct MemoryPool : public RefObject
+    {
```

Comment:
Why have `MemoryPool` be a `RefObject`? Wouldn't the 99% case be that you want a single object to own the memory pool, so that it can include the pool by value (no indirection, allocation, or reference-counting overhead)?